common/components/internal-header/_internal-header.scss

```diff
@@ -1,311 +1,182 @@
-  &::after {
-    @include govuk-shape-arrow($direction: up, $base: 10px, $display: inline-block);
+.app-header__product-name {
+  display: none;
+  @include govuk-font($size: 24);
+
+  @media (min-width: 48.0625em) {
+    display: inline-block;
   }
 }
 
 .app-header__navigation {
-  @include govuk-responsive-margin(2, "bottom");
-  display: block;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-end;
+  list-style: none;
   margin: 0;
   padding: 0;
-  list-style: none;
 
-  @include mq ($from: desktop) {
-    float: right;
+  @media (min-width: 40.0625em) {
+    flex-direction: row;
+    align-items: center;
   }
 }
 
-.js-enabled {
-  .app-header__menu-button {
-    display: block;
-    @include mq ($from: desktop) {
-      display: none;
-    }
-  }
-
-  .app-header__navigation {
-    display: none;
-    @include mq ($from: desktop) {
-      display: block;
-    }
-  }
-
-  .app-header__navigation--open {
-    display: block;
-  }
-}
-
-
-.app-header__navigation--end {
-  @include mq ($from: desktop) {
-    margin: 0;
-    padding: govuk-spacing(1) 0;
-    text-align: right;
-  }
-}
-
-.app-header__navigation--no-service-name {
-  padding-top: govuk-spacing(7);
-}
-
 .app-header__navigation-item {
-  padding: govuk-spacing(2) 0;
-  border-bottom: 1px solid $govuk-header-nav-item-border-color;
+  @include govuk-font($size: 19);
+  margin-bottom: govuk-spacing(1);
+  text-align: right;
 
-  @include mq ($from: desktop) {
-    display: inline-block;
-    margin-right: govuk-spacing(3);
-    padding: govuk-spacing(1) 0;
-    border: 0;
+  @media (min-width: 40.0625em) {
+    margin-right: govuk-spacing(4);
+    padding-right: govuk-spacing(4);
+    margin-bottom: 0;
+    border-right: 1px solid #fff;
   }
 
   a {
-    @include govuk-font($size: 16, $weight: bold);
-    white-space: nowrap;
+    display: inline-block;
   }
-}
-
-.app-header__navigation-item--active {
-  a {
-    &:link,
-    &:hover,
-    &:visited {
-      color: $govuk-header-link-active;
-    }
 
-    // When focussed, the text colour needs to be darker to ensure that colour
-    // contrast is still acceptable
-    &:focus {
-      color: $govuk-focus-text-colour;
-    }
+  &:last-child {
+    margin-right: 0;
+    border-right: 0;
+    padding-right: 0;
   }
 }
 
-.app-header__navigation-item:last-child {
-  margin-right: 0;
-}
-
-@include govuk-media-query($media-type: print) {
-  .app-header {
-    border-bottom-width: 0;
-    color: govuk-colour("black");
-    background: transparent;
-  }
-
-  // Hide the inverted crown when printing in browsers that don't support SVG.
-  .app-header__logotype-crest-fallback-image {
-    display: none;
-  }
-
-  .app-header__link {
-    &:link,
-    &:visited {
-      color: govuk-colour("black");
-    }
+.app-header__link--service-name {
+  height: 100%;
+  align-items: center;
+  display: flex;
+  margin-right: govuk-spacing(4);
+  padding-right: govuk-spacing(4);
+  border-right: 1px solid #fff;
```

Comment:
Are we able to clean up the use of hardcoded colours from these definitions. Maybe we can use one of the variables defined at the start of this file?